reflect/src/main/java/dagger/reflect/ReflectiveDependencyParser.java

```diff
@@ -21,9 +24,11 @@ static void parse(Class<?> cls, Object instance, Scope.Builder scopeBuilder) {
         Type type = method.getGenericReturnType();
         Key key = Key.of(qualifier, type);
 
-        Binding binding = new LinkedProvidesBinding<>(instance, method, NO_BINDINGS);
-
-        scopeBuilder.addBinding(key, binding);
+        if (!alreadySeen.contains(key)) {
```

Comment:
super nit / pet peeve:

you can do
```java
if (alreadySeen.add(key)) {
  // ...
}
```
to avoid having to do double hashing / lookup.

Reply:
Done 👍